cli/file_watcher.rs

```diff
@@ -53,33 +53,40 @@ impl Stream for Debounce {
       inner.event_detected.store(false, Ordering::Relaxed);
       Poll::Ready(Some(()))
     } else {
-      let _ = inner.interval.poll_tick(cx);
-      Poll::Pending
+      match Pin::new(&mut inner.delay).poll(cx) {
+        Poll::Ready(_) => {
+          inner.delay = delay_for(DEBOUNCE_INTERVAL_MS);
+          Poll::Pending
+        }
+        Poll::Pending => Poll::Pending,
+      }
     }
   }
 }
 
-async fn error_handler(watch_future: WatchFuture) {
+async fn error_handler(watch_future: WatchFuture<()>) {
   let result = watch_future.await;
   if let Err(err) = result {
     let msg = format!("{}: {}", colors::red_bold("error"), err.to_string(),);
     eprintln!("{}", msg);
   }
 }
 
-pub async fn watch_func<F>(
-  paths: &[PathBuf],
-  closure: F,
+pub async fn watch_func<F, G>(
+  target_resolver: F,
+  operation: G,
+  job_name: &str,
```

Comment:
Those arguments could use some comments, it's not immediately obvious what's their purpose

Reply:
I added some comments to describe watcher functions including their parameters. PTAL 8feed3c